python/llm/test/benchmark/check_results.py

```diff
@@ -20,26 +20,44 @@
 import sys
 import argparse
 import pandas as pd
+from omegaconf import OmegaConf
 
 def main():
     parser = argparse.ArgumentParser(description="check if the number of lines in html meets expectation")
-    parser.add_argument("-n", "--expected_lines", type=int, dest="expected_lines",
-                        help="the number of expected html lines", default=0)
-    parser.add_argument("-f", "--folder_path", type=str, dest="folder_path",
-                        help="The directory which stores the .csv files", default="./")
+    parser.add_argument("-c", "--csv_name", type=str, dest="csv_name",
+                        help="name of csv", default="")
+    parser.add_argument("-y", "--yaml_name", type=str, dest="yaml_name",
+                        help="name of yaml", default="")
     args = parser.parse_args()
 
-    csv_files = []
-    for file_name in os.listdir(args.folder_path):
-        file_path = os.path.join(args.folder_path, file_name)
-        if os.path.isfile(file_path) and file_name.endswith(".csv"):
-            csv_files.append(file_path)
-    csv_files.sort()
-    
-    number_of_expected_lines=args.expected_lines
-    num_rows = len(pd.read_csv(csv_files[0], index_col=0))
-    if num_rows!=number_of_expected_lines:
-        raise ValueError("The number of arc perf test results does not match the expected value. Please check carefully.")
+    csv_name = args.csv_name
+    file_list = [file for file in os.listdir() if file.endswith('.csv') and csv_name in file]
+    csv_dataframe = pd.read_csv(file_list[0], index_col=0)
+    num_already_test_cases = len(csv_dataframe)
+    already_test_cases = []
+    for index, row in csv_dataframe.iterrows():
+        already_test_cases.append(row['model'] + ":" + row['input/output tokens'].split('-')[0])
+
+    yaml_name = args.yaml_name
+    conf = OmegaConf.load(yaml_name)
+    all_test_cases = []
+    for model in conf.repo_id:
+        for in_out in conf['in_out_pairs']:
+            model_id_input = model + ':' + in_out.split('-')[0]
+            all_test_cases.append(model_id_input)
+
+    exclude_test_cases = []
+    if 'exclude' in conf and conf['exclude'] is not None:
+        exclude_test_cases = conf['exclude']
+
+    num_expected_test_cases = len(all_test_cases) - len(exclude_test_cases)
+
+    if num_already_test_cases != num_expected_test_cases:
+        print("---------------The test cases should be tested!------------")
+        for test_case in all_test_cases:
+            if test_case not in already_test_cases and test_case not in exclude_test_cases:
+                print(test_case)
+        raise ValueError("The above test cases should be tested! Please check the saved csv carefully! ! !")
```

Comment:
The above tests failed. Please check the errors in the log.

Reply:
ok